### PR TITLE
Coverage for: dataRetriever

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unittests for dataRetriever

## Testing procedure
- unit tests only

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: dataRetriever/shardedData/shardedData_test.go [hunk 1]
```diff
@@ -308,3 +312,50 @@ func TestShardedData_SearchFirstDataFoundShouldRetResults(t *testing.T) {
 }
 
 // TODO: Add high load test, reach maximum capacity and inspect RAM usage. EN-6735.
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/shardedData/shardedData_test.go]: removed

File: dataRetriever/requestHandlers/requesters/requesters_test.go [hunk 2]
```diff
@@ -81,6 +82,7 @@ func testRequestDataFromHashArray(t *testing.T, requesterType requestHandlerType
 	t.Run("should work", func(t *testing.T) {
 		t.Parallel()
 
+		_ = logger.SetLogLevel("*:TRACE") // coverage
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/requestHandlers/requesters/requesters_test.go]: removed

**Review by iulianpascalau — DISMISSED** [review]

**Review by bogdan-rosianu — COMMENTED** [review]
typos on the following lines:
- dataRetriever/factory/containers/resolversContainer_test.go:295
- dataRetriever/factory/containers/resolversContainer_test.go:272
- dataRetriever/factory/containers/resolversContainer_test.go:254
- dataRetriever/shardedData/shardedData_test.go:231
- dataRetriever/shardedData/shardedData_test.go:81
- dataRetriever/resolvers/validatorInfoResolver_test.go:382
- dataRetriever/interface.go:319
- dataRetriever/topicSender/diffPeerListCreator_test.go:78
- dataRetriever/factory/epochProviders/currentEpochProvidersFactory_test.go:28
- dataRetriever/mock/marshalizerMock.go:35

File: dataRetriever/storageRequesters/headerRequester_test.go [hunk 3]
```diff
@@ -324,32 +324,73 @@ func TestHeaderRequester_RequestDataFromNonceShouldWork(t *testing.T) {
...
+				assert.Fail(t, "should not have been called")
+				return make([]byte, 0), nil
+			},
+		}
+		hdReq, _ := NewHeaderRequester(arg)
+
+		err := hdReq.RequestDataFromEpoch(epochIdentifier)
+		assert.Equal(t, core.ErrInvalidIdentifierForEpochStartBlockRequest, err)
+	})
+	t.Run("identifier not found should error should error", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/storageRequesters/headerRequester_test.go]: pushed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]
